azurelinuxagent/common/rdma.py

```diff
@@ -94,13 +94,17 @@ def get_rdma_version(self):  # pylint: disable=R1710
             logger.error(error_msg % driver_info_source)
             return
 
-        f = open(driver_info_source)  # pylint: disable=C0103
+        f = open(driver_info_source, "rb")  # pylint: disable=C0103
         while True:
             key = f.read(kvp_key_size)
             value = f.read(kvp_value_size)
             if key and value:
-                key_0 = key.split("\x00")[0]
-                value_0 = value.split("\x00")[0]
+                key_0 = key.partition(b"\x00")[0]
+                if key_0:
+                    key_0 = key_0.decode()
+                value_0 = value.partition(b"\x00")[0]
+                if value_0:
+                    value_0 = value_0.decode()
                 if key_0 == "NdDriverVersion":
                     f.close()
```

Comment:
This comment does not directly related to your changes, but could you please move the file closing to either a `finally` clause or move the file operations to under a `with` clause. This is to ensure that the file always gets closed even if there is some exception thrown while processing the data. 

Sample usage: https://cmdlinetips.com/2016/01/opening-a-file-in-python-using-with-statement/

Thanks!